tests/Command/VoucherCreateCommandTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace App\Tests\Command;
+
+use App\Command\VoucherCreateCommand;
+use App\Creator\VoucherCreator;
+use App\Entity\User;
+use App\Entity\Voucher;
+use App\Enum\Roles;
+use App\Exception\ValidationException;
+use App\Repository\UserRepository;
+use App\Repository\VoucherRepository;
+use App\Validator\Constraints\VoucherUser;
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Exception\UserNotFoundException;
+use Symfony\Component\Validator\ConstraintViolation;
+
+class VoucherCreateCommandTest extends TestCase
```

Comment:
You could also utilize the KernTestCase for that: https://symfony.com/doc/current/console.html#testing-commands

Reply:
Interesting. Gave this a quick test, but it seems like this would require more changes. Since it boots the kernel and uses an actual sqlite database, we'd have to fill this database beforehands.

It would also mean a bit of a paradigm change in how we use unit tests. So far, they're stateless. With this change, unit tests would actually run integration tests. I'm not opposed to this, but so far we have behat for this.

So in any case a bigger topic.